dev/src/reference.ts

```diff
@@ -375,13 +376,16 @@ export class DocumentReference<T = DocumentData> implements Serializable {
       .then(([writeResult]) => writeResult);
   }
 
+  set(data: Partial<T>, options: SetOptions): Promise<WriteResult>;
+  set(data: T, options?: SetOptions): Promise<WriteResult>;
   /**
    * Writes to the document referred to by this DocumentReference. If the
    * document does not yet exist, it will be created. If you pass
    * [SetOptions]{@link SetOptions}, the provided data can be merged into an
-   * existing document.
+   * existing document. Using Partial objects requires
+   * [SetOptions]{@link SetOptions} to be passed in as well.
```

Comment:
I feel this detail is confusion. We should expand on this in an additional paragraph or omit it.

Reply:
Omitted since we already explain it in `toFirestoreFromMerge`